packages/jest-cli/src/TestRunner.js

```diff
@@ -59,7 +59,7 @@ class TestRunner {
     options: Options,
   ) {
     this._config = config;
-    this._dispatcher = new ReporterDispatcher();
+    this._dispatcher = new ReporterDispatcher({hasteContext: hasteMap});
```

Comment:
Can we instead make this so we pass `hasteContext.instance` (also the variable shouldn't be hasteMap in the constructor, that is an oversight :) ). That way you can simply type it as `HasteMap` and it will be easier for me to refactor when I take on the haste map work I'm planning to do.